wweval/R/get_epidemic_phases_from_rt.R

```diff
@@ -86,3 +86,29 @@ get_epidemic_phases_from_rt <- function(locations,
 
   return(df_epi_phase)
 }
+
+#' Get the order of the categorical horizons for plotting
+#'
+#' @param df a dataframe containing the column name `phase` which will
+#' contain character strings indicating the phase of the epidemic
+#' (increasing, decreasing, peak, or nadir) of  the score/quantile/sample.
+#'
+#' @return a dataframe containing the same columns as `df` but with the
+#' addition of `fig_order` and reordered in terms of `fig_order` for plotting.
+#' @export
+order_phases <- function(df) {
+  df_w_order <- df |>
+    dplyr::mutate(
+      fig_order = dplyr::case_when(
+        phase == "increasing" ~ 1,
+        phase == "decreasing" ~ 2,
+        phase == "peak" ~ 3,
+        phase == "nadir" ~ 4,
+        TRUE ~ 5
+      )
+    ) |>
+    dplyr::mutate(
+      horizon = forcats::fct_reorder(phase, fig_order)
+    )
```

Comment:
As above, I think you don't need `forcats::fct_reorder` here:

```suggestion
  phase_order <- c(
      "increasing",
      "decreasing",
      "peak",
      "nadir")
  df_w_order <- df |>
    dplyr::mutate(phase = factor(phase, ordered = TRUE, levels = phase_order)
```

Reply:
Related: are you sure you want the order to be  `"increasing", "decreasing", "peak", "nadir"`? I might suggest `"increasing", "peak", "decreasing", "nadir"` or some wraparound of that (e.g.  `"decreasing", "nadir", "increasing", "peak"`)

Reply:
Changed